### PR TITLE
Fix versioning and many formatting/null check

### PR DESCRIPTION
**Pull request by mwpreston**
This PR fixes the versioning for the report as well as implements many null checks on the protected object section.

Much feedback from beta users has been applied to the code now.  

Also applies basic formatting and structure around the changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,38 @@
 # Rubrik CDM As Built Report Changelog
 
+## Unreleased
+
+### Modified
+
+* Null checks on backup sources
+
```
Comment (tpcarman) [inline CHANGELOG.md]: Update release information & include date e.g. `## [0.0.7] - YYYY-MM-dd`

File: README.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ Below are the instructions on how to install, configure, and generate a Rubrik C
 ## Pre-requisites
 The following PowerShell modules are required for generating a Rubrik As Built report.
 
-Each of these modules can be easily downloaded and installed via the PowerShell Gallery 
+Each of these modules can be easily downloaded and installed via the PowerShell Gallery
 
-- [Rubrik CDM Powershell SDK](https://www.powershellgallery.com/packages/Rubrik/) 
+- [Rubrik CDM Powershell SDK](https://www.powershellgallery.com/packages/Rubrik/)
     *Note:* The Rubrik As Built Report Module currently relies on the `devel` branch of the Rubrik PowerShell SDK. Follow the instructions below to ensure the `devel` branch is the active installation.
```
Comment (tpcarman) [inline README.md]: Is this still required now that the module is publish to the PS Gallery?

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 [hunk 3]
```diff
@@ -1,19 +1,21 @@
 function Invoke-AsBuiltReport.Rubrik.CDM {
     <#
-    .SYNOPSIS  
+    .SYNOPSIS
         PowerShell script to document the configuration of Rubrik CDM in Word/HTML/XML/Text formats
     .DESCRIPTION
         Documents the configuration of the Rubrik CDM in Word/HTML/XML/Text formats using PScribo.
     .NOTES
-        Version:        0.0.2
+        Version:        0.0.6
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: Does this need updating to `0.0.7`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
+## [0.0.6]
```
Comment (tpcarman) [inline CHANGELOG.md]: Add date `- YYYY-MM-dd`

**Review by tpcarman — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mwpreston) [reply CHANGELOG.md]: Done

File: README.md (hunk 2 above)
Comment (mwpreston) [reply README.md]: No - updated docs to show that

File: Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1 (hunk 3 above)
Comment (mwpreston) [reply Src/Public/Invoke-AsBuiltReport.Rubrik.CDM.ps1]: updated all to reflect new version

File: CHANGELOG.md (hunk 4 above)
Comment (mwpreston) [reply CHANGELOG.md]: and done

**Review by tpcarman — APPROVED** [review]

**Pull request merged**